client/store/userInput.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * ACTION TYPES
+ */
+const FIELD_EDIT_ADDPRODUCT_NAME = 'FIELD_EDIT_ADDPRODUCT_NAME';
+const FIELD_EDIT_ADDPRODUCT_PRICE = 'FIELD_EDIT_ADDPRODUCT_PRICE';
+const FIELD_EDIT_ADDPRODUCT_DESC = 'FIELD_EDIT_ADDPRODUCT_DESC';
+const FIELD_EDIT_ADDPRODUCT_IMAGEURL = 'FIELD_EDIT_ADDPRODUCT_IMAGEURL';
+/**
+ * INITIAL STATE
+ */
+
+const initialState = {
+  addProduct: {
+    name: {
+      value: '',
+      untouched: true,
+    },
+    price: {
+      value: '0',
+      untouched: true,
+    },
+    description: {
+      value: '',
+      untouched: true,
+    },
+    imageUrl: {
+      value: '',
+      untouched: true,
+    },
+  },
+};
```

Comment:
Hmm. So I don't think this is really best practice in respect to modularity. I think I understand why you're using a nested object here, but I think you should make a slice specifically for the `addProduct` input fields which would be an object and if we need another similar slice we just make another slice.

Also you make want to take a look at Redux Form, since it seems like they simplify handling form input. I mean, it does actually seem like they use nested objects. But I just think in our case it's best to keep them separate, because [nested objects in Redux](https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns) just add more complexity that I just don't think we need to deal with.